src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -157,7 +159,7 @@ public void teleportHome(@NotNull Player player, @NotNull Island island, int del
      */
     private void teleportHome(@NotNull Player player, @NotNull Island island) {
         player.setFallDistance(0);
-        player.teleport(LocationUtils.getSafeLocation(island.getHome(), island));
+        PaperLib.teleportAsync(player, LocationUtils.getSafeLocation(island.getHome(), island), PlayerTeleportEvent.TeleportCause.PLUGIN);
```

Comment:
Please try async teleports with spigot versions may taco tho

Reply:
teleportAsync dont exist with spigot api, only Paper

Reply:
no i mean if user is uses spigot this method will work with PaperLib#SpigotEnvironment, u should try a spigot version tho